dcurve/src/internal/merkle.ts

```diff
@@ -0,0 +1,98 @@
+import { parseUnits } from 'ethers/lib/utils';
+import {
+  MerkleDistributorInfo,
+  NewFormat,
+  parseBalanceMap,
+} from '@dgrants/utils/src/merkle-distributor/parse-balance-map';
+import { PayoutMatch } from '../types';
+
+// export MerkleDistributorInfo
+export type { MerkleDistributorInfo };
+
+/**
+ * Generates hash of the GrantsDistribution distribution
+ * @param distribution - PayoutMatch[]
+ * @param currDecimals - The number of decimals used by the distributed token
+ * @returns {MerkleDistributorInfo} - merkle tree of the distribution
+ * https://github.com/Uniswap/merkle-distributor
+ */
+export const generateMerkle = (distribution: PayoutMatch[], currDecimals: number) => {
+  // collect the input
+  const merkleInput: NewFormat[] = [];
+
+  // foreach payout parse the earnings
+  distribution.forEach((payoutMatch: PayoutMatch) => {
+    merkleInput.push({
+      address: payoutMatch.address,
+      earnings: parseUnits(payoutMatch.match.toString(), currDecimals).toHexString(),
```

Comment:
We should clearly document (in the README?) that the input values in `payoutMatch` are human-readable values, and not parsed based on decimals, since this deviates from the typical interface

Reply:
Was this added somewhere

Reply:
I've added this to the readme now 👍 (in 1016d54)